plugins/woocommerce/src/Admin/PluginsHelper.php

```diff
@@ -532,4 +534,57 @@ public static function get_activation_status( $job_id = null ) {
 		return self::get_action_data( $actions );
 	}
 
+	/**
+	 * Show notices to connect to woo.com for unconnected store in the plugin list.
+	 *
+	 * @return void
+	 */
+	public static function maybe_show_connect_notice_in_plugin_list() {
+		if ( 'woocommerce_page_wc-settings' !== get_current_screen()->id ) {
+			return;
+		}
+
+		if ( WC_Helper::is_site_connected() ) {
+			return;
+		}
+
+		$notice_string = '';
+
+		if ( count( WC_Helper::get_local_woo_plugins() ) > 0 ) {
+			$notice_string .= __( 'Your store might be at risk as you are running old versions of Woo plugins.', 'woocommerce' );
+			$notice_string .= ' ';
+		}
+
+		$connect_page_url = add_query_arg(
+			array(
+				'page' => 'wc-admin',
+				'tab'  => 'my-subscriptions',
+				'path' => rawurlencode( '/extensions' ),
+			),
+			admin_url( 'admin.php' )
+		);
+
+		$notice_string .= sprintf(
+			/* translators: %s: Connect page URL */
+			__( '<a href="%s">Connect your store</a> to Woo.com to get updates and streamlined support for your subscriptions.', 'woocommerce' ),
+			esc_url( $connect_page_url )
+		);
+		echo '<script type="text/javascript">
```

Comment:
default wp `is-dismissible` only adds an `x` close button but it does not persisted, we store the dismissal information in the local storage (similar to an existing approach we use in the admin react component)